api/src/main/java/io/hyperfoil/impl/Util.java

```diff
@@ -89,6 +90,12 @@ public static String toString(ByteBuf buf, int offset, int length) {
       }
    }
 
+   public static AsciiString toAsciiString(ByteBuf buf, int offset, int length) {
+      final byte[] bytes = new byte[length];
```

Comment:
I'm not sure about memory allocations in `static` methods. In most cases a preallocated memory segment can be used, and it can be "memset(asciiBuffer, 0, ...)" if necessary.

Since `length` is a parameter here - we could check against "sizeof(asciiBuffer)" and only allocate if needed.

malloc(), memset() and free() has a massive overhead in high performance application, and for Java that means GC as well. So, the fewer we can do the better

Reply:
the `static` context shouldn't change the mechanism used by JIT to perform allocations, but just the so-called "escape context": as long as something "escape" (meaning that the ref of allocated memory go out of scope to be used elsewhere) a real allocation is required, which magnitude depends by few JVM parameters (alignments, COOPS, etc etc) and where (TLAB/yng gen or old gen).

Reply:
Yeah, but what is the improvement of using 

```
private static byte[] ASCII_BUFFER = new byte[ASCII_BUFFER_SIZE];
```

on a class level instead ?

Reply:
Class level one require synchronization, but in this specific case there are no temporary copies (that's the reason of the change).

In the original code a tmp `byte[]` was allocated and then passed to the JDK lib's `String`, that will internally use it to produce the internal `byte[]/char[]` holding the content of the `String`, while here we create a single `byte[]` and  passing it into a freshly (empty) `AsciiString` to be used as content (without any copy).

In total we had:
- one byte[] + one String (with it's internal byte[]/char[])

while here:
- one byte[] + one AsciiString

we're both saving a useless copy and allocation

Reply:
Lets target `master` only

Reply:
I don't have labelling powers yet